projects/packages/publicize/src/class-publicize-setup.php

```diff
@@ -26,6 +29,30 @@ public static function configure() {
 		add_action( 'jetpack_feature_publicize_enabled', array( __CLASS__, 'on_jetpack_feature_publicize_enabled' ) );
 	}
 
+	/**
+	 * Initialization of publicize logic that should always be loaded.
+	 */
+	public static function pre_initialization() {
+
+		$is_wpcom = ( new Host() )->is_wpcom_simple();
+
+		// Assets are to be loaded in all cases.
+		Publicize_Assets::configure();
+
+		$rest_controllers = array(
+			Connections_Controller::class,
+		);
+
+		// Load the REST controllers.
+		foreach ( $rest_controllers as $controller ) {
+			if ( $is_wpcom ) {
+				wpcom_rest_api_v2_load_plugin( $controller );
+			} else {
+				new $controller();
+			}
+		}
```

Comment:
This should really be part of `on_jetpack_feature_publicize_enabled`, which is where we're initialising the other REST API elements, as well as registering the site options, which we'll need if we're bringing those features to WPCOM.

We could move this in there and then call `on_jetpack_feature_publicize_enabled`, but that would register the Connections_Post_Field and REST_Controller on WPCOM, as well as other things that might conflict.

What do you think about calling `pre_initialization` in `on_jetpack_feature_publicize_enabled` and then only using the pattern in actions.php for WPCOM? If we did that, we might want to rename the `pre_initialization` method or something.

Reply:
Let us improve that initialization in a follow up PR

Reply:
I have created a task for it and added it to our backlog.